pages/agent/v3/signed-pipelines.md

```diff
@@ -0,0 +1,114 @@
+  In short, it's because RSASSA-PKCS1 v1.5 signatures are generally regarded to be less secure than the newer RSA-PSS signatures. While RSASSA-PKCS1 v1.5 signatures are still largely known to be relatively secure, we want to encourage our users to use the most secure algorithms possible, so when using RSA keys, we only support RSA-PSS signatures. We also recommend looking into ECDSA and EdDSA signatures, which are generally regarded to be more secure than RSA signatures.
+</details>
+
+### What algorithm should I use?
+
+Signed Pipelines supports a number of different signing algorithms, which can largely be broken down into two categories: Symmetric Signing Algorithms (where signing and verification use the same keys), and Asymmetric Signing Algorithms (where signing and verification use different keys).
+
+We recommend using an asymmetric signing algorithm, as it creates a permissions boundary between the agents that upload and sign pipelines, and agents that run other jobs and have the ability to verify signatures. This means that were an attacker able to compromise instances with verification keys, they wouldn't be able to modify the steps that are run on your agents.
+
+With regards to specific algorithm choice, basically any of the asymmetric signing algorithms we support are fine and will be secure. If you're not sure which one to use, `EdDSA` is proven to be secure, has a modern design, wasn't designed by a Nation State Actor, and produces nice short signatures.
+
+Signed pipelines supports the following signing algorithms:
+
+#### Symmetric
+- `HS256`
+- `HS384`
+- `HS512`
+
+#### Asymmetric
+- `PS256`
+- `PS384`
+- `PS512`
+- `ES256`
+- `ES384`
+- `ES512`
+- `EdDSA`
+
+### Making your agents sign steps they upload 
+
+Now that your keys are generated, we need to configure your agents to use them. On agents that will be uploading pipelines, add the following to your agent's config file:
+```ini
+job-signing-jwks-path=<path to signing keys>
+signing-key-id=<the key id you generated earlier>
+job-verification-jwks-path=<path to verification keys>
+```
+
+On instances that will verify jobs, add:
+```ini
+job-verification-jwks-path=<path to verification keys>
+```
+
+No
+
+### Signing the pipeline's stored configuration
+
+[ed: this section feels clunky to me, and i'm not sure the best way to explain it - we need to sign the initial steps of a pipeline as well as the steps that get uploaded by an agent, but it's difficult to concisely convey all the concepts involved]
+
+In Buildkite, pipelines have two YAML definitions - one that's stored Buildkite, which you can change on the settings page for a given pipeline, and other(s) that are uploaded by agents. In this guide, we'll refer to the former as the "stored" configuration, and the latter as "uploaded" configuration. The stored configuration generally contains the initial steps needed to run a pipeline, and almost always contains a step that uploads a pipeline to Buildkite.
```

Comment:
Perhaps something like 
```suggestion
Builds in Buildkite pipelines are initialized using steps that are configured on the pipeline itself. In the course of running a job, the Buildkite Agent can [upload further steps to the pipeline](link to some info about this elsewhere in the docs). We will refer to the former as "configured steps" and the latter as "uploaded steps". It is common for the configured steps to be a single step that uploads the remainder of pipeline from [a YAML file in the repository](link to info about ./buildkite.yaml).
```

Reply:
"To ensure that all steps in a pipeline are signed, you will need to manually sign the configured steps. To do this..." etc

Reply:
ooooooh love it